src/Microsoft.OData.Edm/Validation/EdmErrorCode.cs

```diff
@@ -1388,67 +1388,57 @@ public enum EdmErrorCode
         /// </summary>
         AnnotationApplyToNotAllowedAnnotatable = 400,
 
-        #region JSON
+        /// <summary>
+        /// Invalid $Key value.
+        /// </summary>
+        InvalidKeyValue,
 
         /// <summary>
-        /// Invalid JSON
+        /// Invalid mulitple entity container in one schema
         /// </summary>
-        JsonInvalid,
+        SchemaCannotHaveMoreThanOneEntityContainer,
 
         /// <summary>
-        /// The specified version number is not valid.
+        /// Invalid $Collection value.
         /// </summary>
-        JsonInvalidVersionNumber,
+        InvalidCollectionValue,
+
+        #region JSON
 
         /// <summary>
-        /// Invalid JSON value
+        /// Invalid JSON
         /// </summary>
-        JsonInvalidValue,
+        InvalidJson,
 
         /// <summary>
         /// Invalid JSON value kind
         /// </summary>
-        JsonUnknownElementValueKind,
+        UnknownJsonElementValueKind,
 
         /// <summary>
         /// Invalid JSON number type
         /// </summary>
-        JsonInvalidNumberType,
+        InvalidJsonNumberType,
 
         /// <summary>
         /// Encountered a JSON element that is not supported
         /// </summary>
-        JsonUnsupportedElement,
+        UnsupportedJsonElement,
 
         /// <summary>
         /// Encountered a JSON element that was never used
         /// </summary>
-        JsonUnexpectedElement,
+        UnexpectedJsonElement,
 
         /// <summary>
         /// Excountered a JSON elemnt that was not expected.
         /// </summary>
-        JsonUnexpectedValueKind,
+        UnexpectedJsonValueKind,
 
         /// <summary>
         /// A JSON property is required in an JSON object
         /// </summary>
-        JsonMissingRequiredProperty,
-
-        /// <summary>
-        /// Invalid $Collection value.
-        /// </summary>
-        JsonInvalidCollectionValue,
-
-        /// <summary>
-        /// Invalid $Key value.
-        /// </summary>
-        JsonInvalidKeyValue,
-
-        /// <summary>
-        /// Invalid mulitple entity container in one schema
-        /// </summary>
-        JsonSchemaCannotHaveMoreThanOneEntityContainer
+        MissingRequiredJsonProperty,

```

Comment:
>MissingRequiredJsonProperty [](start = 8, length = 27)

Can this be something like "MissingRequiredAttribute"?  isn't the important thing to the application the fact that something wasn't specified that needed to be (not that it was represented as a JSON property)?